client/my-sites/email/email-management/home/email-plan.jsx

```diff
@@ -133,10 +135,24 @@ class EmailPlan extends React.Component {
 		}
 
 		if ( hasTitanMailWithUs( domain ) ) {
-			const hasSubscriptionId = !! getTitanSubscriptionId( domain );
+			if ( getTitanSubscriptionId( domain ) ) {
+				return {
+					path: emailManagementNewTitanAccount( selectedSite.slug, domain.name, currentRoute ),
+				};
+			}
+
+			const showExternalControlPanelLink = ! isEnabled( 'titan/iframe-control-panel' );
+			const controlPanelUrl = showExternalControlPanelLink
+				? emailManagementTitanControlPanelRedirect( selectedSite.slug, domain.name, currentRoute, {
+						context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+				  } )
+				: emailManagementManageTitanAccount( selectedSite.slug, domain.name, currentRoute, {
+						context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+				  } );
```

Comment:
Not mandatory at all but we could do the following in order to avoid duplicate code:

```javascript
const getControlPanelUrl = ( pathFunction ) => pathFunction( selectedSite.slug, domain.name, currentRoute, {
  context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL
} );

const controlPanelUrl = getControlPanelUrl( showExternalControlPanelLink ? emailManagementTitanControlPanelRedirect : emailManagementManageTitanAccount );
```

Reply:
For now, I'll leave this as-is -- I know I end up searching for `functionName(` every now and then to find usages.